### PR TITLE
Personal/yankun/library packing configuration

### PR DESCRIPTION
**Pull request by yankunhuang-pku**
## Description

Export the conversion related function (`convert`) to pack a library, which will be used in the VS Code extension

## Testing

used `npm publish` to publish it and have used it in a VS Code extension, which can work effectively.

### CODE REVIEW

File: src/lib/workers/workerUtils.js [hunk 1]
```diff
@@ -6,13 +6,16 @@
...
 module.exports.workerTaskProcessor = function (taskProcessor) {
-    parentPort.on('message', (msg) => {
-        taskProcessor(msg.msg)
-            .then(data => { 
-                msg.channelWorkerPort.postMessage(data);
-            }).catch(err => {
-                msg.channelWorkerPort.postMessage((err instanceof Error) ? err.message : err);
-            });
-    });
+    if (parentPort){
```
Comment (moria97) [inline src/lib/workers/workerUtils.js:9]: do we need handle else case here?

Comment (yankunhuang-pku) [reply src/lib/workers/workerUtils.js:9]: Good question~  @deepak-ms   Can you help us take a look at it? I'm not sure if I need to handle else case here, and just keep the old behavior.

Comment (yankunhuang-pku) [reply src/lib/workers/workerUtils.js:9]: I don't use the workerPool when packing this lib now. It seems that we don't need to handle else and when the parentPort is null, we just keep the old behavior and don't run the taskProcessor.

File: src/lib/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+var convert = require('./workers/conversion').convert;
+
+module.exports = {
+    convert: convert
```
Comment (ACMoretxj) [inline src/lib/index.js:4]: Maybe "{ convert }" could be simpler than "{ convert: convert }" if the external name is the same with the internal name.

Comment (BoyaWu10) [reply src/lib/index.js:4]: Just curious why is this file named index.js?

Comment (ACMoretxj) [reply src/lib/index.js:4]: > Just curious why is this file named index.js?

From my perspective, the index.js filename is a convention as a function hub of a lib. And there is a syntactic sugar that `require('@/package/lib')` is equal to `require('@/package/lib/index.js')`.

Comment (yankunhuang-pku) [reply src/lib/index.js:4]: Thanks, I will change it "{ convert }".
I think, although the filename 'index.js' can be named arbitrarily, it generally defaults to index.js in node.js, which can be seen as the index of the exported variables and functions.

Comment (BoyaWu10) [reply src/lib/index.js:4]: Got it. Thanks!

**Pull request closed without merging**